src/bors/handlers/pr_events.rs

```diff
@@ -753,4 +761,81 @@ mod tests {
         })
         .await;
     }
+
+    #[sqlx::test]
+    async fn delete_completed_auto_build_on_push(pool: sqlx::PgPool) {
+        BorsBuilder::new(pool)
+            .github(gh_state_with_merge_queue())
+            .run_test(async |tester| {
+                tester.post_comment("@bors r+").await?;
+                tester.expect_comments(1).await;
+                tester.process_merge_queue().await;
+                tester.expect_comments(1).await;
+                tester
+                    .wait_for_default_pr(|pr| pr.auto_build.is_some())
+                    .await?;
+                tester.workflow_full_success(tester.auto_branch()).await?;
+                tester.expect_comments(1).await;
+                tester
+                    .push_to_pr(default_repo_name(), default_pr_number())
+                    .await?;
+                tester.expect_comments(1).await;
+                tester
+                    .wait_for_default_pr(|pr| pr.auto_build.is_none())
+                    .await?;
+                Ok(())
+            })
+            .await;
+    }
+
+    #[sqlx::test]
+    async fn cancel_pending_auto_build_on_push_comment(pool: sqlx::PgPool) {
+        BorsBuilder::new(pool)
+            .github(gh_state_with_merge_queue())
+            .run_test(async |tester| {
+                tester.post_comment("@bors r+").await?;
+                tester.expect_comments(1).await;
+                tester.process_merge_queue().await;
+                tester.expect_comments(1).await;
+                tester
+                    .wait_for_default_pr(|pr| pr.auto_build.is_some())
+                    .await?;
+                tester.workflow_start(tester.auto_branch()).await?;
+                tester
+                    .push_to_pr(default_repo_name(), default_pr_number())
+                    .await?;
+                insta::assert_snapshot!(tester.get_comment().await?, @r"
```

Comment:
This won't actually work for rollups, because the rollup PR won't be pushed to :thinking: But I haven't ever seen this occur in practice anyway.